javatest: extra to integrate java unit testing into standard waf_unit_test environment

fedepell opened the pull request:
Compared to the code in the java demo this integrates with the waf_unit_test so should be much easier to integrate with other languages unit tests (ie. pytest). It is also more versatile for different test environment and was testes with TestNG and JUnit.

ita1024 commented:
The examples that you are trying to add are distributed under the Apache 2 license and you are not the author, so we cannot merge this.

fedepell commented:
Sorry for that. Let me rewrite the examples.

Does the rest look more or less fine?

Thanks!

ita1024 commented:
I believe so. If you are making changes, please fix the mixed indentation so that it is tabs-only.

fedepell commented:
I created a new test from scratch, fixed the indentation (sorry for that) and a few minor doc details. I squashed it all together to clean up.

Hope it's fine now!
